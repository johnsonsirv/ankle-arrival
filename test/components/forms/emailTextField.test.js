/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable react/react-in-jsx-scope */
/* eslint-disable no-undef */
import EmailTextField from '../../../src/components/forms/inputs/emailTextField';

function setup() {
  const props = {
    name: 'email',
    onChange: jest.fn(),
  };
  const enzymeWrapper = mount(<EmailTextField {...props} />);

  return { props, enzymeWrapper };
}

describe('<EmailTextField /> rendering', () => {
  const { enzymeWrapper } = setup();
  it('renders correctly', () => {
    expect(enzymeWrapper).toMatchSnapshot();
  });
  it('should render 1 editable <input> email', () => {
    expect(enzymeWrapper.find('input[type="email"]')).toHaveLength(1);
  });
});

describe('<EmailTextField /> interactions', () => {
  const { props, enzymeWrapper } = setup();
  it('should handle onChange', () => {
    enzymeWrapper.simulate('change', {
      target: { value: 'example@example.com' },
    });
    expect(props.onChange).toHaveBeenCalled();
  });
});
