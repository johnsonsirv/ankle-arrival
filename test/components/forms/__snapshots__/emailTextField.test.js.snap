// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmailTextField /> rendering renders correctly 1`] = `ReactWrapper {}`;
